sources/packages/backend/apps/api/src/route-controllers/institution-locations/_tests_/e2e/institution-location.students.controller.getOptionsList.e2e-spec.ts

```diff
@@ -0,0 +1,75 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import * as request from "supertest";
+import {
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  FakeStudentUsersTypes,
+  getStudentToken,
+} from "../../../../testHelpers";
+import {
+  E2EDataSources,
+  createE2EDataSources,
+  saveFakeDesignationAgreementLocation,
+} from "@sims/test-utils";
+
+describe("InstitutionLocationStudentsController(e2e)-getOptionsList", () => {
+  let app: INestApplication;
+  let db: E2EDataSources;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+    // Before running the test, making the existing location as non designated.
+    await db.designationAgreementLocation.update(
```

Comment:
I'm not sure if we should do an update like this. It could interfere with other tests.

Reply:
I think that for the API e2e tests we're not doing that. Those tests are for queue-consumers  only, right?

Reply:
ya, the current test files are integration files. But does it matter? Do you have any other suggestion other than a db update, in this scenario?

Reply:
The API tests are not running sequentially as the ones from `queue-consumers`, which is why it is ok for queue-consumers and not for the API.
The current idea for API is that tests should create specific scenarios as needed.

Reply:
As a suggestion, a new location can be created, since it is not part of any designation agreement it will be considered as not designated and it will not be returned to the list.
If I am not wrong, the new location can also be added to an existing designation and have the `approved` column set to `false` on `sims.designation_agreement_locations`.